iter.go

```diff
@@ -9,6 +9,9 @@ import (
 // Query is the function used to get a page listing.
 type Query func(*form.Values) ([]interface{}, ListMeta, error)
 
+// Query2 is the function used to get a page listing.
+type Query2 func(*Params, *form.Values) ([]interface{}, ListMeta, error)
```

Comment:
As with `Call2` and `GetIter2`, `Query2` becomes `Query` as soon as we finish refactoring everything. It'll never hit `master`.